src/segmentation.py

```diff
@@ -0,0 +1,45 @@
+import bpy
+import bpycv
+import numpy as np
+import cv2
+from typing import Dict
+
+
+class Segmentation:
+    def __init__(self, classes: Dict[int, int] = {}) -> None:
+        """
+        classes: a dictionary of class ids to segmentation map values
+        """
+        self.classes = classes
+
+    def add_class(self, class_id: int, segmentation_value: int):
+        self.classes[class_id] = segmentation_value
+
+    def remove_class(self, class_id: int):
+        del self.classes[class_id]
+
+    def segment(self, output_file: str):
+        self._assign_classes()
+        rendered_data = self._render_segmentation()
+        segmentation = rendered_data["inst"]
+        self._write_segmentation(segmentation, output_file)
+
+    def _assign_classes(self):
+        for obj in bpy.data.objects:
+            if "segmentation_id" in obj:
```

Comment:
wondering if these if statements can be combined

Reply:
Ideally not. Not all objects have segmentation_id. I need to check that first